### PR TITLE
Modified signature validation: get only signer addresses on challenge

### PR DESCRIPTION
**Pull request by tomaszslabon**
This PR modifies validation of DKG result during challenge: instead of getting addresses of all operators from sortition-pool, we just get addresses of the operators that signed the result. 

When running tests in `RandomBeacon.GroupCreation.test.ts` the average gas consumption for `challengeDkgResult` decreased from `1201940` to `1160869`.

### CODE REVIEW

File: solidity/random-beacon/contracts/DKGValidator.sol [hunk 1]
```diff
@@ -209,16 +209,21 @@ contract DKGValidator {
             )
         );
 
-        address[] memory membersAddresses = sortitionPool.getIDOperators(
-            result.members
+        uint32[] memory signerIds = new uint32[](
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply solidity/random-beacon/contracts/DKGValidator.sol:212]: Also curious if it wouldn't be a bit cheaper if we assigned `uint256[] signingMembersIndices = result.signingMembersIndices;` at the beginning:
```
        uint256[] signingMembersIndices = result.signingMembersIndices;
        uint32[] memory signingMemberIds = new uint32[](signingMembersIndices);
        for (uint256 i = 0; i < result.signingMembersIndices.length; i++) {
            signingMemberIds[i] = result.members[signingMembersIndices[i] - 1];
        }

        address[] memory signingMembersAddresses = sortitionPool.getIDOperators(
            signingMemberIds
        );
```

File: solidity/random-beacon/contracts/DKGValidator.sol [hunk 2]
```diff
@@ -209,16 +209,23 @@ contract DKGValidator {
...
+        uint32[] memory signingMemberIds = new uint32[](
+            result.signingMembersIndices.length
+        );
+        for (uint256 i = 0; i < result.signingMembersIndices.length; i++) {
+            signingMemberIds[i] = result.members[
+                result.signingMembersIndices[i] - 1
+            ];
+        }
+
+        address[] memory signersAddresses = sortitionPool.getIDOperators(
```
Comment (dimpar) [inline solidity/random-beacon/contracts/DKGValidator.sol:221]: signersAddresses -> signingMemberAddresses? Same comment for line 237

**Comment by dimpar** [comment]
no more comments

File: solidity/random-beacon/contracts/DKGValidator.sol (hunk 2 above)
Comment (tomaszslabon) [reply solidity/random-beacon/contracts/DKGValidator.sol]: Done in https://github.com/keep-network/keep-core/pull/2745/commits/000b4bce8772a6e22211166aaf9b7fef0f91dcd1.

**Review by dimpar — APPROVED** [review]
lgtm. I'll wait with merging once we know if the gas goes down based on Piotr's comment.

File: solidity/random-beacon/contracts/DKGValidator.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply solidity/random-beacon/contracts/DKGValidator.sol]: @pdyraga The gas consumption fell after this change from `1160547` to `1147950` for `challengeDkgResult`.
I committed it in https://github.com/keep-network/keep-core/pull/2745/commits/249a6aaeb92bb661db212c3c702f77ff270fdba0.

[the remark this answers is not in the document]
Comment (nkuba) [reply solidity/random-beacon/contracts/DKGValidator.sol]: > 

Interesting... Good to know!

**Review by nkuba — APPROVED** [review]

File: solidity/random-beacon/contracts/DKGValidator.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply solidity/random-beacon/contracts/DKGValidator.sol]: @pdyraga Also when I assigned signatures to call data variable:
```
bytes calldata signatures = result.signatures;
```

The gas consumption fell further to `1142118`, so it fell about `5000`. Should I commit that as well?

**Pull request merged**